fuel-vm/src/interpreter/balances.rs

```diff
@@ -91,14 +92,12 @@ impl RuntimeBalances {
         self.state.get(asset).map(Balance::value)
     }
 
-    fn set_memory_balance_inner(balance: &Balance, memory: &mut [u8; MEM_SIZE]) -> Result<Word, RuntimeError> {
+    fn set_memory_balance_inner(balance: &Balance, memory: &mut VmMemory) -> Result<Word, RuntimeError> {
```

Comment:
If we are sure that `write_bytes` can't fail, then let's change the signature of the method too

```suggestion
    fn set_memory_balance_inner(balance: &Balance, memory: &mut VmMemory) ->  Word {
```